src/lib.rs

```diff
@@ -256,35 +256,30 @@ pub mod contract {
                     return Err(InvariantError::PriceLimitReached);
                 }
 
-                // TODO: refactor
-                let mut tick = Tick::default();
-
-                let update_limiting_tick = limiting_tick.map(|(index, bool)| {
-                    if bool {
-                        tick = self.ticks.get(pool_key, index).unwrap();
-                        (index, Some(&mut tick))
-                    } else {
-                        (index, None)
+                if let Some((tick_index, is_initialized)) = limiting_tick {
+                    if is_initialized {
+                        let mut tick = self.ticks.get(pool_key, tick_index).unwrap();
+                        let has_crossed = pool.cross_tick(
+                            result,
+                            swap_limit,
+                            tick_index,
+                            &mut tick,
+                            &mut remaining_amount,
+                            by_amount_in,
+                            x_to_y,
+                            current_timestamp,
+                            &mut total_amount_in,
```

Comment:
return new `total_amount_in` instead of mutating it inside function. This is a bad practice